### PR TITLE
fix(python): `README.md` snippet builder fix; avoid silent failure during config generation.

### PR DESCRIPTION
**Pull request by eyw520**
…process.

@coderabbitai summary

### CODE REVIEW

File: generators/python/src/fern_python/generators/sdk/sdk_generator.py [hunk 1]
```diff
@@ -285,10 +285,13 @@ def run(
...
                         generated_root_client=generated_root_client,
                     )
-                except Exception:
+                except Exception as e:
                     generator_exec_wrapper.send_update(
                         GeneratorUpdate.factory.log(
-                            LogUpdate(level=LogLevel.DEBUG, message=f"Failed to generate README.md; this is OK")
+                            LogUpdate(
+                                level=LogLevel.DEBUG,
+                                message=f"Failed to generate README.md with exception {e}; this is OK",
```
Comment (dsinghvi) [inline generators/python/src/fern_python/generators/sdk/sdk_generator.py:293]: Up to make this user facing message a bit nicer: 

`"Failed to generate README.md..."`

File: generators/python/src/fern_python/generator_cli/readme_snippet_builder.py [hunk 2]
```diff
@@ -62,38 +62,42 @@ def __init__(
...
             return [self._endpoint_snippet_map[endpoint_id].sync_client for endpoint_id in specified_endpoint_ids]
 
-        # If no endpoints are specified, return the an endpoint that uses the feature
         endpoints_with_feature = self._filter_endpoint_ids_by_feature(feature_id)
-        if len(endpoints_with_feature) > 0:
-            return [self._endpoint_snippet_map[endpoints_with_feature[0]].sync_client]
+        filtered_endpoints_with_feature = [e for e in endpoints_with_feature if e in self._endpoint_snippet_map]
+
+        if len(filtered_endpoints_with_feature) > 0:
+            return [self._endpoint_snippet_map[filtered_endpoints_with_feature[0]].sync_client]
```
Comment (dsinghvi) [inline generators/python/src/fern_python/generator_cli/readme_snippet_builder.py:100]: What happens if `self._endpoint_snippet_map[filtered_endpoints_with_feature[0]]` returns `None` ?

File: generators/python/src/fern_python/generator_cli/readme_snippet_builder.py [hunk 3]
```diff
@@ -62,38 +62,42 @@ def __init__(
...
-        if usage_endpoint_ids is not None:
-            return [self._endpoint_snippet_map[endpoint_id].sync_client for endpoint_id in usage_endpoint_ids]
-        return [self._endpoint_snippet_map[self._default_endpoint_id].sync_client]
+        try:
+            usage_endpoint_ids = self._get_user_specified_endpoint_ids_for_feature(
+                feature_id=ReadmeSnippetBuilder.USAGE_FEATURE_ID
+            )
+            if usage_endpoint_ids is not None:
+                return [self._endpoint_snippet_map[endpoint_id].sync_client for endpoint_id in usage_endpoint_ids]
+            return [self._endpoint_snippet_map[self._default_endpoint_id].sync_client]
```
Comment (dsinghvi) [inline generators/python/src/fern_python/generator_cli/readme_snippet_builder.py:86]: What happens if `self._endpoint_snippet_map[self._default_endpoint_id]` returns None ?

**Review by dsinghvi — APPROVED** [review]
🚀

File: generators/python/src/fern_python/generators/sdk/sdk_generator.py (hunk 1 above)
Comment (eyw520) [reply generators/python/src/fern_python/generators/sdk/sdk_generator.py]: Improved.

**Pull request merged**